packages/plugins/response-cache/src/plugin.ts

```diff
@@ -32,18 +32,16 @@ type Context = {
 /**
  * Function for building the response cache key based on the input parameters
  */
-export type BuildResponseCacheKeyFunction = (params: {
-  /** Raw document string as sent from the client. */
-  documentString: string;
-  /** Variable values as sent form the client. */
-  variableValues: ExecutionArgs['variableValues'];
-  /** The name of the GraphQL operation that should be executed from within the document. */
-  operationName?: Maybe<string>;
-  /** optional sessionId for make unique cache keys based on the session.  */
-  sessionId?: Maybe<string>;
-}) => Promise<string>;
-
-export type GetDocumentStringFromContextFunction = (params: DefaultContext) => Maybe<string>;
+export type BuildResponseCacheKeyFunction = (
+  params: {
+    /** Raw document string as sent from the client. */
+    documentString: string;
+    /** optional sessionId for make unique cache keys based on the session.  */
+    sessionId?: Maybe<string>;
+  } & ExecutionArgs
+) => Promise<string>;
+
```

Comment:
Currently we did not supply the context here, because we wanted this function to be as portable as possible to the edge/proxy that runs before the GraphQL API in order to short circuit.

Since most of the `ExecutionArgs` parameters are optional (`contextValue`, `variableValues`), I think it is fine for those. 

However, `document` might not be present on a "edge/proxy" level as it would apply this logic on the raw GraphQL payload. It's value should be optional.

I don't think passing `fieldResolver`, `typeResolver` or `subscribeFieldResolver` to this function provides any value.

Reply:
I agree with the `defaultGetDocumentString ` API change. However, the `BuildResponseCacheKeyFunction` signature should be reverted to its previous implementation as I don't see a benefit in it.

Reply:
Done!